### ISSUE TITLE
'contractSize' from Bitget is wrong.

### ISSUE THREAD

**Issue by kjnh10** [ccxt/ccxt#16157]
- OS: Linux
- Programming Language version: Python3.9
- CCXT version: 2.4.43

```
import ccxt
exchange = ccxt.bitget()
exchange.load_markets()
print(exchange.markets['MTL/USDT:USDT']['contractSize'])

```

```
0.1
```

The returned value should be None or 1 as it was before version 2.1.28.

**Comment by ttodua** [comment]
@samgermain the last time bitget seems updated by you, do you remind what might be clue? you might have more insights about that change. if you can't handle this, I'll take.

**Comment by kjnh10** [comment]
@ttodua 
Hi, is there any update on this?

**Comment by kjnh10** [comment]
@ttodua
Hi, is there any update on this?

**Comment by kjnh10** [comment]
It seems 'contractSize' is set from 'sizeMultiplier'. But this is not apporopriate.
<img width="607" alt="image" src="https://user-images.githubusercontent.com/6933428/214009403-a8af968e-d39f-4ebd-a575-e7c206b5794f.png">

By the document by bitget, 'sizeMultiplier' is something like unit number.
<img width="468" alt="image" src="https://user-images.githubusercontent.com/6933428/214009229-5f81b9eb-c19e-4332-9bfe-18cbfdbf2012.png">
https://bitgetlimited.github.io/apidoc/en/mix/#get-all-symbols

**Comment by kjnh10** [comment]
If you think this is not bug, could you tell me the definition of 'contractSize'?

I'm using 'contractSize' for adjusting the amount to be passed for create_order(), because the amount I really trade is different from the amount to be passed to create_order() in some exchanges like OKX.

**Comment by kjnh10** [comment]
By the ccxt document, 'contractSize' definition is below.
<img width="665" alt="image" src="https://user-images.githubusercontent.com/6933428/214010882-cf795648-1ffd-40c9-801b-dc91ff5da0d2.png">

So I think this is a bug as I expected.

**Comment by ttodua** [comment]
@samgermain do you have some insights about this?

**Comment by kjnh10** [comment]
@samgermain 
Thanks for the reply.

I almost agree with you regarding this.
>
>The contract size specifies the amount of the underlying asset that 1 contract is equal to in a contract market
>For example, in a market like ADA/USDT:USDT if the contract size is 10, and the ticker price is 1 USDT, then when >you purchase 1 contract within the ADA/USDT:USDT (using createOrder("ADA/USDT:USDT", "market", "buy", 1)this will cost you 10 USDT

Below, I will use MTL/USDT:USDT for example, because 'contractSize' of ADA/USDT:USDT is actually 1 which is not bug. ('contractSize' of MTL/USDT:USDT is 0.1 and this is bug)


In Bitget, createOrder("MTL/USDT:USDT", "market", "buy", 1) will cost you 1 USDT, if the price of MTL is 1 USDT.
This means the contract size of "MTL/USDT:USDT" should be 1, though ccxt returns 'contractSize' as 0.1. 

In summary, 'sizeMultiplier' of bitget is not the contract size as you think.

**Comment by kjnh10** [comment]
@ttodua @samgermain 
Is there any update?
I think this is a just tiny simple bug.

**Comment by samgermain** [comment]
> @samgermain Thanks for the reply.
> 
> I almost agree with you regarding this.
> 
> > The contract size specifies the amount of the underlying asset that 1 contract is equal to in a contract market
> > For example, in a market like ADA/USDT:USDT if the contract size is 10, and the ticker price is 1 USDT, then when >you purchase 1 contract within the ADA/USDT:USDT (using createOrder("ADA/USDT:USDT", "market", "buy", 1)this will cost you 10 USDT
> 
> Below, I will use MTL/USDT:USDT for example, because 'contractSize' of ADA/USDT:USDT is actually 1 which is not bug. ('contractSize' of MTL/USDT:USDT is 0.1 and this is bug)
> 
> In Bitget, createOrder("MTL/USDT:USDT", "market", "buy", 1) will cost you 1 USDT, if the price of MTL is 1 USDT. This means the contract size of "MTL/USDT:USDT" should be 1, though ccxt returns 'contractSize' as 0.1.
> 
> In summary, 'sizeMultiplier' of bitget is not the contract size as you think.

I'm just trying to understand how to verify this. Can you supply some examples with code?

**Comment by dmunozfer** [comment]
> > @samgermain Thanks for the reply.
> > I almost agree with you regarding this.
> > > The contract size specifies the amount of the underlying asset that 1 contract is equal to in a contract market
> > > For example, in a market like ADA/USDT:USDT if the contract size is 10, and the ticker price is 1 USDT, then when >you purchase 1 contract within the ADA/USDT:USDT (using createOrder("ADA/USDT:USDT", "market", "buy", 1)this will cost you 10 USDT
> > 
> > 
> > Below, I will use MTL/USDT:USDT for example, because 'contractSize' of ADA/USDT:USDT is actually 1 which is not bug. ('contractSize' of MTL/USDT:USDT is 0.1 and this is bug)
> > In Bitget, createOrder("MTL/USDT:USDT", "market", "buy", 1) will cost you 1 USDT, if the price of MTL is 1 USDT. This means the contract size of "MTL/USDT:USDT" should be 1, though ccxt returns 'contractSize' as 0.1.
> > In summary, 'sizeMultiplier' of bitget is not the contract size as you think.
> 
> I'm just trying to understand how to verify this. Can you supply some examples with code that prove what you're talking about?

Let me try to explain. If I have a contract size of 10SBTC per contract and I want to open a position of 1SBTC, I should open an order of 0.1 contracts. If I have a contract size of 0.1SBTC and I want to open a position of 1SBTC, I should open an order of 10 contracts.

I have checked the code: ContractSize is retrieved from the sizeMultiplier field. According to the bitget documentation: "sizeMultiplier: Quantity Multiplier The order size must be greater than minTradeNum and equal to the multiple of sizeMultiplier". I think this refers to the ccxt value of "precision.amount".
